src/_gettsim/shared.py

```diff
@@ -385,3 +385,100 @@ def format_key_path(key_tuple: tuple[str, ...]) -> str:
                     raise TypeError(msg)
 
     _assert_valid_gettsim_pytree(tree, current_key=())
+
+
+def get_group_by_id_path(
+    target_path: tuple[str],
+    group_by_functions_tree: NestedFunctionDict,
+) -> tuple[str] | None:
+    """Get the group_by_id for an aggregation target.
+
+    The group_by_id is the id of the group over which the aggregation is performed. If
```

Comment:
```suggestion
    The group by-identifier is the id of the group over which the aggregation is performed. If
```
Same in the rest here and in other docstrings. We don't want to document a variable that is only used in the function body.